File: ios-app/SharedPackage/Sources/SharedKit/Models/TranscriptionEntry.swift
import Foundation

public struct TranscriptionEntry: Identifiable, Codable, Equatable, Sendable {
    public let id: UUID
    public let timestamp: Date
    public let durationSeconds: Double
    public var text: String?
    public var rawText: String?
    public var status: TranscriptionStatus
    public var audioFilePath: String?
    public var errorMessage: String?

    public init(
        id: UUID = UUID(),
        timestamp: Date = Date(),
        durationSeconds: Double,
        text: String? = nil,
        rawText: String? = nil,
        status: TranscriptionStatus = .recording,
        audioFilePath: String? = nil,
        errorMessage: String? = nil
    ) {
        self.id = id
        self.timestamp = timestamp
        self.durationSeconds = durationSeconds
        self.text = text
        self.rawText = rawText
        self.status = status
        self.audioFilePath = audioFilePath
        self.errorMessage = errorMessage
    }
}

File: android-app/app/src/main/java/com/whisperapp/domain/model/TranscriptionEntry.kt
package com.whisperapp.domain.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "transcription_entries")
data class TranscriptionEntry(
    @PrimaryKey val id: String,
    val timestamp: Long,
    val durationSeconds: Double,
    val rawText: String?,
    val text: String?,
    val status: TranscriptionStatus,
    val audioFilePath: String?,
    val errorMessage: String?
)
